pywbem_mock/_mainprovider.py

```diff
@@ -19,10 +19,14 @@
 #
 
 """
-Implementation of for a CIM repository that processes WBEM server requests
-destined for the CIMRepository.
-
-For documentation, see mocksupport.rst.
+Implementation of a CIM general provider that processes:
+    The CIMClass request Operations.
+    The CIMQualifier request Operations
+    The CIMInstance request Enumerate and request pull Operations
+    The Associator request operations
+
+This provider uses a separate datastore for the objects maintained by
+the repository and defined with the interfaces in _base_repository.py
 """
```

Comment:
Currently, the `MainProvider` class has some of the instance methods that should be moved to `BaseInstanceProvider`. I think `BaseInstanceProvider` should also be part of this PR. Then you can also add the `InvokeMethod` method there.